### ISSUE TITLE
Traning error

### ISSUE THREAD

**Issue by lixiangnlp** [facebookresearch/fairseq-lua#8]
I have re-installed torch and dependency libraries. But I met the thread errors when training on my own training data. Although the beginning seems correctly. I trained the model on 2 GPU cards. The training command is as follow.

fairseq train -sourcelang src -targetlang ref -datadir nist-bin/ -model blstm -nhid 512 -dropout 0.2 -dropout_hid 0 -optim adam -lr 0.0001 -savedir data/blstm -batchsize 128 -maxbatch 70 -validbleu -nembed 128

The error is : 

| [ref] Dictionary: 40003 types
| [src] Dictionary: 40003 types
| IndexedDataset: loaded data/ with 127615 examples
| IndexedDataset: loaded data/ with 88 examples
| IndexedDataset: loaded data/ with 164 examples
| IndexedDataset: loaded data/ with 88 examples
| IndexedDataset: loaded data/ with 164 examples
| epoch 000 | 0001000 updates | words/s     685| trainloss    10.67 | train ppl  1627.17
| epoch 000 | 0002000 updates | words/s     879| trainloss     9.95 | train ppl   987.71
| epoch 000 | 0003000 updates | words/s     668| trainloss     9.68 | train ppl   821.81
/home/lixiang/library/dnn/torch/install/bin/luajit: ...rary/dnn/torch/install/share/lua/5.1/threads/threads.lua:183: [thread 2 callback] ...rary/dnn/torch/install/share/lua/5.1/nngraph/gmodule.lua:348: split(2) cannot split 0 outputs
stack traceback:
	[C]: in function 'error'
	...rary/dnn/torch/install/share/lua/5.1/nngraph/gmodule.lua:348: in function 'neteval'
	...rary/dnn/torch/install/share/lua/5.1/nngraph/gmodule.lua:380: in function 'forward'
	.../torch/install/share/lua/5.1/rnnlib/nn/SequenceTable.lua:150: in function 'forward'
	.../torch/install/share/lua/5.1/rnnlib/nn/SequenceTable.lua:150: in function 'uo'
	.../torch/install/share/lua/5.1/rnnlib/recurrentnetwork.lua:54: in function 'func'
	...rary/dnn/torch/install/share/lua/5.1/nngraph/gmodule.lua:345: in function 'neteval'
	...rary/dnn/torch/install/share/lua/5.1/nngraph/gmodule.lua:380: in function 'func'
	...rary/dnn/torch/install/share/lua/5.1/nngraph/gmodule.lua:345: in function 'neteval'
	...rary/dnn/torch/install/share/lua/5.1/nngraph/gmodule.lua:380: in function 'forward'
	...hare/lua/5.1/fairseq/torchnet/ResumableDPOptimEngine.lua:356: in function <...hare/lua/5.1/fairseq/torchnet/ResumableDPOptimEngine.lua:333>
	[C]: in function 'xpcall'
	...rary/dnn/torch/install/share/lua/5.1/threads/threads.lua:234: in function 'callback'
	...ibrary/dnn/torch/install/share/lua/5.1/threads/queue.lua:65: in function <...ibrary/dnn/torch/install/share/lua/5.1/threads/queue.lua:41>
	[C]: in function 'pcall'
	...ibrary/dnn/torch/install/share/lua/5.1/threads/queue.lua:40: in function 'dojob'
	[string "  local Queue = require 'threads.queue'..."]:13: in main chunk
stack traceback:
	[C]: in function 'error'
	...rary/dnn/torch/install/share/lua/5.1/threads/threads.lua:183: in function 'dojob'
	...rary/dnn/torch/install/share/lua/5.1/threads/threads.lua:264: in function 'synchronize'
	...hare/lua/5.1/fairseq/torchnet/ResumableDPOptimEngine.lua:371: in function 'doTrain'
	...hare/lua/5.1/fairseq/torchnet/ResumableDPOptimEngine.lua:189: in function 'train'
	...nn/torch/install/share/lua/5.1/fairseq/scripts/train.lua:405: in main chunk
	[C]: in function 'require'
	...rch/install/lib/luarocks/rocks/fairseq/scm-1/bin/fairseq:17: in main chunk
	[C]: at 0x00405d50

**Comment by jgehring** [comment]
I think you ended up with a batch that is too small, maybe because there's a sentence longer then 140 words (maxbatch is 70, times two GPUs). Can you try removing the `-maxbatch` flag? This should not be required for blstms since the mini-batch is limited by `-bptt` already (which is set to 25 by default). The flag is useful for controlling memory usage when training with `-bptt 0` (e.g. for fully convolutional models). If a the number of tokens (number of sentences * length of longest target sentence) in the mini-batch is larger than `-maxbatch`, the mini-batch will be split along the batch dimension (as opposed to the time dimension with `-bptt`).

**Comment by jgehring** [comment]
Sorry for the delay; has this been resolved?

**Comment by jgehring** [comment]
Closing the issue now, please re-open if necessary.
